<launch>
  <!-- app master -->
  <arg name="master" default="true" doc="launch master if enabled"/>
  <arg name="master_address" default="localhost" doc="address for app_manager master"/>
  <arg name="master_port" default="11313" doc="port for app_manager master"/>

  <!-- app list -->
  <arg name="use_applist" default="false" doc="load apps from applist argument"/>
  <arg name="applist" default="" doc="app dirs (space separated)"/>

  <!-- robot_name -->
  <arg name="use_robot_name" default="false" doc="load robot_name from argument"/>
  <arg name="robot_name" default="" doc="robot name"/>

  <!-- robot_type -->
  <arg name="use_robot_type" default="false" doc="load robot_type from argument"/>
  <arg name="robot_type" default="" doc="robot type"/>

  <!-- respawn -->
  <arg name="respawn" default="false" doc="respawn app_manager"/>

  <!-- internal use -->
  <arg name="applist_args" default="--applist $(arg applist)" if="$(arg use_applist)"/>
  <arg name="applist_args" default=""                     unless="$(arg use_applist)"/>
  <arg name="robot_name_args" default="--robot-name $(arg robot_name)"
       if="$(arg use_robot_name)"/>
  <arg name="robot_name_args" default="" 
       unless="$(arg use_robot_name)"/>
  <arg name="robot_type_args" default="--robot-type $(arg robot_type)"
       if="$(arg use_robot_type)"/>
  <arg name="robot_type_args" default="" 
       unless="$(arg use_robot_type)"/>

  <node pkg="app_manager" type="appmaster" name="appmaster"
        args="-p $(arg master_port)" if="$(arg master)" respawn="$(arg respawn)"/>
  <node pkg="app_manager" type="app_manager" name="app_manager"
        args="$(arg applist_args) $(arg robot_name_args) $(arg robot_type_args)"
        output="screen" respawn="$(arg respawn)">
    <param name="interface_master" value="http://$(arg master_address):$(arg master_port)"/>
  </node>
</launch>
